File: level3/네트워크.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

bool visit[200] = {false};

void dfs(int start, vector<vector<int>> computers, int n) {
    visit[start] = true;
    for(int i=0; i<n; i++){
        if(!visit[i] && computers[start][i])
            dfs(i, computers, n);
    }
}

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;

    for(int i=0; i<n; i++){
        if(!visit[i]){
            dfs(i, computers, n);
            answer++;
        }
    }
    return answer;
}

File: JavaScript/level3/네트워크.js
function solution(n, computers) {
    var answer = 0;
    let visited = Array(n).fill(false);
    
    function dfs(location){
        visited[location] = true;
        for(let i=0; i<n; i++)
            if(computers[location][i] === 1 && i !== location && visited[i] === false)
                dfs(i);
    }   
    
    for(let i=0; i<n; i++)
        if(visited[i] === false){
            dfs(i);
            answer++;
        }
    
    return answer;
}